Maestro.Tests Compiler Error

Maestro.Tests fails to compile

**Steps to reproduce**
1. Generate o3de.sln
2. Build Maestro.Tests

**Expected behavior**
All modules compile

**Actual behavior**
Compile error
```
Building Custom Rule D:/prj/o3de/Gems/Maestro/Code/CMakeLists.txt
D:\prj\o3de\Code\Legacy\CryCommon\Mocks\ICryPakMock.h(78,5): error C2275: 'testing::internal::Function<AZ::IO::IArchive::LevelPackOpenEvent *(void)>::ArgumentTuple': expected an expression instead of a type [D:\prj\o3de\build\windows\External\Maestro-4dedec36\Code\Maestro.Tests.
vcxproj]
      MOCK_METHOD0(GetLevelPackOpenEvent, auto()->LevelPackOpenEvent*);
      ^ (compiling source file 'CMakeFiles/Maestro.Tests.dir/Unity/unity_0_cxx.cxx')
      D:\prj\o3de\Code\Legacy\CryCommon\Mocks\ICryPakMock.h(78,5):
      prefix the qualified-id with 'typename' to indicate a type
          MOCK_METHOD0(GetLevelPackOpenEvent, auto()->LevelPackOpenEvent*);
```


**Found in Branch**
development
fe4c6e969856918ffc0f0ba743c2d671928def93 
Mon Sep 16 11:17:35 2024 -0700

**Desktop/Device (please complete the following information):**
 - OS: Windows
 - MSBuild version 17.11.2

--- a/Code/Legacy/CryCommon/Mocks/ICryPakMock.h
+++ b/Code/Legacy/CryCommon/Mocks/ICryPakMock.h
@@ -75,8 +75,8 @@ struct CryPakMock
     MOCK_CONST_METHOD0(GetPakPriority, AZ::IO::FileSearchPriority());
     MOCK_CONST_METHOD1(GetFileOffsetOnMedia, uint64_t(AZStd::string_view szName));
     MOCK_CONST_METHOD1(GetFileMediaType, EStreamSourceMediaType(AZStd::string_view szName));
-    MOCK_METHOD0(GetLevelPackOpenEvent, auto()->LevelPackOpenEvent*);
-    MOCK_METHOD0(GetLevelPackCloseEvent, auto()->LevelPackCloseEvent*);
+    MOCK_METHOD0(GetLevelPackOpenEvent, LevelPackOpenEvent*());
+    MOCK_METHOD0(GetLevelPackCloseEvent, LevelPackCloseEvent*());
 
 };
 

